migration_helper_iterator.go

```diff
@@ -0,0 +1,132 @@
+package anser
+
+import (
+	"context"
+	"sync"
+
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/anser/client"
+	"github.com/mongodb/anser/model"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type cursorMigrationMetadataIterator struct {
+	cursor  client.Cursor
+	catcher grip.Catcher
+	item    *model.MigrationMetadata
+	ctx     context.Context
+}
+
+func (c *cursorMigrationMetadataIterator) Err() error {
+	c.catcher.Add(c.cursor.Err())
```

Comment:
I still don't love that `Err()` assumes we call it once and I don't understand why we can't just use `c.catcher.Add()` in all places where `c.cursor` is used (i.e. `Next()` and `Close()`).